File: 17_ORM & Code Structure/Task 01/models/package.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const Package = sequelize.define('Package', {
  id: {
    type: DataTypes.INTEGER,
    autoIncrement: true,
    primaryKey: true,
  },
  name: {
    type: DataTypes.STRING,
    allowNull: false,
  },
  sender: {
    type: DataTypes.STRING,
    allowNull: false,
  },
  receiver: {
    type: DataTypes.STRING,
    allowNull: false,
  },
  sender_location: {
    type: DataTypes.STRING,
    allowNull: false,
  },
  recipient_location: {
    type: DataTypes.STRING,
    allowNull: false,
  },
  fee: {
    type: DataTypes.INTEGER,
    allowNull: false,
  },
  weight: {
    type: DataTypes.FLOAT,
    allowNull: false,
  },
});

module.exports = Package;

File: 18_middleware/task_api/models/package.go
package models

import (
	"github.com/jinzhu/gorm"
)

type Package struct {
	gorm.Model
	Name             string  `json:"name"`
	Sender           string  `json:"sender"`
	Receiver         string  `json:"receiver"`
	SenderLocation   string  `json:"sender_location"`
	ReceiverLocation string  `json:"receiver_location"`
	Fee              float64 `json:"fee"`
	Weight           float64 `json:"weight"`
}
